testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/batch/deployment/JobXmlVisibilityTestCase.java

```diff
@@ -216,7 +215,7 @@ private static JavaArchive createEjb(final String name) {
         final JavaArchive jar = ShrinkWrap.create(JavaArchive.class, name)
                 .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
                 // Add at least one EJB annotated class to make this an EJB JAR
-                .addClasses(CountingItemReader.class, CountingItemWriter.class, BatchExecutionService.class);
+                .addClasses(CountingItemReader.class, CountingItemWriter.class, SimpleEJB.class);
```

Comment:
@jamezp I realize now that I forget to exclude this test at https://github.com/wildfly/wildfly/pull/13016/files#diff-ec12efbf5a13385ba26a3162a8b61e72R1190 but it still passes. It runs against a server with no EJB and still passes. It's not a surprise the deployment doesn't fail because of no EJB, as having no subsystem means the SimpleEJB is not processed. I just don't know whether the test is still meaningful; i.e. should I exclude it?

Reply:
I'll need to debug it to see what's going on here. I _think_ the point of this was to ensure it was processed as a sub-deployment. Since this is only used in an EAR that is probably why it works. I do need to test to see what happens when the EJB JAR is in a WAR.

That said I'm comfortable doing whatever you think is appropriate here. I will look into the issue I described above, but I think that's separate from what you're doing. TBH I'm fine with removing the `SimpleEJB` all together as I'm not sure what I was thinking when I thought an EJB was required :)

Reply:
Actually the EJB isn't required at all. I think I used 

> Add at least one EJB annotated class to make this an EJB JAR

 because I was using EJB's for the tests. The spec does not require it to be an EJB at all. It can be any JAR. I think we can remove the comment and the `SimpleEJB`. This test is really just testing that the deployment model can see the correct job XML files.

Reply:
After another look I think we need to leave the test as you have it. In other words what you have is what we need.